### PR TITLE
Challenge #10 in Java (Unreviewed)

### PR DESCRIPTION
**Pull request by zmiller91**

### CODE REVIEW

File: challenge_10/java/zmiller91/ValidClosers.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        
+        Stack<Character> stack = new Stack<>();
+        for(char c : input.toCharArray()) {
+            
+            // No rules for openers, add it to the stack
+            if (openers.contains(c)) {
+                stack.push(c);
+            }
+            
+            // Openers have a few rules
```
Comment (moonpatroller) [inline challenge_10/java/zmiller91/ValidClosers.java]: Closers

File: challenge_10/java/zmiller91/ValidClosers.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            if (closers.contains(c)) {
+                
+                // Stack cannot be empty
+                if(stack.empty()) {
+                    System.out.println(FALSE);
```
Comment (moonpatroller) [inline challenge_10/java/zmiller91/ValidClosers.java]: Just do println(false)

File: challenge_10/java/zmiller91/ValidClosers.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                
+                // The last element pushed must be an opener
+                int openerIdx = openers.indexOf(stack.pop());
+                if(openerIdx == -1) {
+                    System.out.println(FALSE);
+                    return;
+                }
+                
+                // The opener and closer must match
+                Character expected = closers.get(openerIdx);
```
Comment (moonpatroller) [inline challenge_10/java/zmiller91/ValidClosers.java]: Just use char

File: challenge_10/java/zmiller91/ValidClosers.java (hunk 2 above)
Comment (zmiller91) [inline challenge_10/java/zmiller91/ValidClosers.java]: I put it as a variable for a reason -- it allows me to change the string to "Sorry, please try again" with ease. It was an intentional design decision.

File: challenge_10/java/zmiller91/ValidClosers.java (hunk 3 above)
Comment (zmiller91) [inline challenge_10/java/zmiller91/ValidClosers.java]: `closers` is a `Character` list not a `char` list. I do not want to cast it.

**Pull request merged**